File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ''
        if len(strs) == 1:
            res = strs[0]
        else:
            for i, char in enumerate(list(strs[0])):
                x = False
                for j in range(1, len(strs)):
                    if len(strs[j]) >= i+1 and strs[j][i] == char:
                        x = True
                    else:
                        x = False
                        break
                if x:
                    res += char
                else:
                    break
        return res

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String s = "";
        if (strs.length == 1) {
            s = strs[0];
        } else if (strs.length > 1) {
            Arrays.sort(strs);
            // System.out.println(Arrays.toString(strs));
            if (!strs[0].isEmpty()) {
                int i = 0;
                while (true) {
                    if (i < strs[0].length() && strs[0].charAt(i) == strs[strs.length - 1].charAt(i)) {
                        s += strs[0].charAt(i);
                        i ++;
                    } else {
                        break;
                    }
                }
            }
        }
        return s;
    }
}